pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test: TestLogic_cursor failed

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_cursor [failed](https://mesolite.cluster.engflow.com/invocations/default/17472f6a-6eec-44ef-b059-f7f65cdd7ecd?testReportRun=5&testReportShard=18&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC12ZWMtb2ZmOmZha2VkaXN0LXZlYy1vZmZfdGVzdA==) on release-26.1 @ [e39600cb891be2624d33bc7d171ada4629abdfdb](https://github.com/cockroachdb/cockroach/commits/e39600cb891be2624d33bc7d171ada4629abdfdb):


```
=== RUN   TestLogic_cursor
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestLogic_cursor901892777
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:123: cluster virtualization disabled in global scope due to issue: #156124 (expected label: C-bug)
[08:59:28] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cursor with config fakedist-vec-off: 191 tests, 1 failures
[08:59:29] --- total progress: 191 statements
--- total: 191 tests, 1 failures
    logic.go:4769: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_cursor901892777
--- FAIL: TestLogic_cursor (4.35s)
=== RUN   TestLogic_cursor/regression
    logic.go:2397: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cursor:852: 
        expected success, but found
        (57014) query execution canceled due to statement timeout
        errors.go:580: in init()
--- FAIL: TestLogic_cursor/regression (1.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=18</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_cursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58981

--- a/pkg/sql/logictest/testdata/logic_test/cursor
+++ b/pkg/sql/logictest/testdata/logic_test/cursor
@@ -848,6 +848,7 @@ SET statement_timeout = '1s';
 DECLARE a CURSOR FOR SELECT * FROM ( VALUES (1), (2) ) t(id);
 
 # Note we can't set pg_sleep to 1 or else it'll statement timeout!
+retry
 statement ok
 select pg_sleep(0.7)
 
@@ -856,6 +857,7 @@ FETCH 1 FROM a
 ----
 1
 
+retry
 statement ok
 select pg_sleep(0.7)
 

